File: examples/node/spec/unit/versionless.spec.ts
"use strict";
import nock from "nock";
import Twilio from "../../lib/rest/Twilio";

describe("versionless", () => {
  const accountSid: string = "AC12345678123456781234567812345678";
  const authToken: string = "CR12345678123456781234567812345678";
  const twilio = new Twilio(accountSid, authToken);

  afterAll(() => {
    nock.cleanAll();
  });

  it("should create a deployed devices fleet", () => {
    const scope = nock("http://versionless.twilio.com")
      .post("/DeployedDevices/Fleets", {
        Name: "Otis",
      })
      .reply(201, { name: "Otis" });

    return twilio.versionless.deployed_devices.fleets
      .create({ name: "Otis" })
      .then((fleetInstance) => {
        expect(fleetInstance.name).toEqual("Otis");
        scope.done();
      });
  });

  it("should fetch a deployed devices fleet", () => {
    const scope = nock("http://versionless.twilio.com")
      .get("/DeployedDevices/Fleets/123")
      .reply(200, { sid: "123", friendly_name: "Friend-o" });

    return twilio.versionless.deployed_devices
      .fleets("123")
      .fetch()
      .then((fleetInstance) => {
        expect(fleetInstance.sid).toEqual("123");
        expect(fleetInstance.friendlyName).toEqual("Friend-o");
        scope.done();
      });
  });

  it("should page the understand assistants", () => {
    const scope = nock("http://versionless.twilio.com")
      .get("/understand/Assistants")
      .reply(200, {
        assistants: [
          {
            friendly_name: "flea",
          },
        ],
        meta: {
          key: "assistants",
        },
      });

    return twilio.versionless.understand.assistants
      .page()
      .then((assistantPage) => {
        expect(assistantPage.instances[0].friendlyName).toEqual("flea");
      })
      .then(() => scope.done());
  });
});

File: examples/ruby/spec/integration/versionless_spec.rb
require 'spec_helper.rb'
describe 'Integration Test' do
  before(:each) do
    @account_sid = 'AC12345678123456781234567812345678'
    @auth_token = 'CR12345678123456781234567812345678'
  end

  it "can make create call to Deployed Devices" do
    @holodeck.mock(Twilio::Response.new(200, '
            {
           "name":"Fleet Name",
           "sid": "AC12345678123456781234567812345678",
          "friendlyName": "Friendly Fleet "
           }
           '))
    createDeployedDevices = @client.versionless.deployed_devices.fleets.create()
    expect(createDeployedDevices.sid).to eq(@account_sid)
    expect(createDeployedDevices.name).to eq('Fleet Name')
  end
end
